plugins/linux-v4l2/CMakeLists.txt

```diff
@@ -1,31 +1,43 @@
-project(linux-v4l2)
+cmake_minimum_required(VERSION 3.22...3.25)
+
+legacy_check()
 
 option(ENABLE_V4L2 "Build OBS with v4l2 support" ON)
+option(ENABLE_UDEV "Build linux-v4l2 with UDEV support" ON)
```

Comment:
The switches are added regardless of the values, the early exit happens _after_ defining the options.

Reply:
Right, I meant "with this change", since before this change `ENABLE_UDEV` is seemingly defined after the early exit. Mostly asking because if this is actually a change, devs who disable `ENABLE_V4L2` will see a spurious `ENABLE_UDEV` enabled, either in CMakeCache or cmake-gui, unless I'm mistaken.

Reply:
Even if we move the option below the check for the entire module, all it takes is one successful configuration with V4L2 enabled for that cache entry to appear and I'm not convinced of putting in the effort to reset it to `OFF` automatically or cleaning it up.

As far as I could tell it is not common practice to fully reproduce dependencies of options with another in the cache, instead values are just quietly ignored if a precondition is not met.